flusher.go

```diff
@@ -215,25 +215,33 @@ func (s *Server) reportGlobalMetricsFlushCounts(ms metricsSummary) {
 }
 
 func (s *Server) flushS3(finalMetrics []DDMetric) {
-	start := time.Now()
+	const delimiter = '\t'
 
-	var data bytes.Buffer
-	err := json.NewEncoder(&data).Encode(finalMetrics)
+	start := time.Now()
+	csv, err := encodeDDMetricsCSV(finalMetrics, delimiter)
 	if err != nil {
 		s.logger.WithFields(logrus.Fields{
 			"metrics": len(finalMetrics),
-		}).Error("Could not marshal finalMetrics before posting to s3")
+		}).Error("Could not marshal finalMetrics before posting to s3: ", err)
```

Comment:
whoops, missed this in the previous PR - should probably pass errors using WithError or ErrorKey, ie one of:

```go
s.logger.WithFields(logrus.FIelds{
    logrus.ErrorKey: err,
    "metrics":       len(finalMetrics),
}).Error("could not marshal...")
```

```go
s.logger.WithField("metrics", len(finalMetrics)).WithError(err).Error("could not marshal...")
```